### PR TITLE
Connect API to Bookstore

### PR DESCRIPTION
**Pull request by Epaltechs**
### In this branch of the project, I have implemented the following:
- Add redux-thunk to the project.
- Use fetch for making API calls.
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch list of books.
- Dispatch another action that pass API response to reducer and updates the state.
- Refactor add book and remove book features to persist changes in the server.
- Change existing Action Creators, so they return functions.
- Make changes by using API.
- Dispatch an action that updates the state accordingly.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,36 +1,45 @@
...
+
 // reducer
 const booksReducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
       return [...state, action.book];
 
     case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.bookId);
+      return state.filter((book) => book.item_id !== action.bookId);
```
Comment (ShahierNashaat) [inline src/redux/books/books.js:43]: - [ ] Kindly be noticed that you should create a new action type to get fetch the books ie: 
```
const FETCH_BOOK = 'bookStore/books/FETCH_BOOK';
```
Screenshot from 2022-06-16 00-21-16

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Epaltechs , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Epaltechs** [comment]
#### As required, in this commit I have created a new action type to fetch the books

**Review by topeogunleye — APPROVED** [review]
### Hi @Epaltechs, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
> 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**